platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/TrackingSettings.java

```diff
@@ -280,36 +276,13 @@ void resetTrackingModesIfRequired(boolean translate, boolean rotate) {
   /**
    * Reset the tracking modes as necessary. Animated camera position changes can reset the underlying tracking modes.
    *
-   * @param cameraPosition the changed camera position
+   * @param currentCameraPosition the current camera position
+   * @param targetCameraPosition  the changed camera position
+   * @param isFromLocation        true if from location
    */
-  void resetTrackingModesIfRequired(CameraPosition cameraPosition) {
-    if (isDismissTrackingModesForCameraPositionChange()) {
-      resetTrackingModesIfRequired(cameraPosition.target != null, false);
-    }
-  }
-
-  /**
-   * Returns if a animation allows to dismiss a tracking mode.
-   * <p>
-   * By default this is set to true.
-   * </p>
-   *
-   * @return True if camera animations will allow to dismiss a tracking mode
-   */
-  public boolean isDismissTrackingModesForCameraPositionChange() {
-    return isResetTrackingWithCameraPositionChange;
-  }
-
-  /**
-   * Sets a flag to allow animated camera position changes to dismiss a tracking mode.
-   * <p>
-   * <p>
-   * </p>
-   *
-   * @param willAllowToDismiss True will allow animated camera changes dismiss a trackig mode
-   */
-  public void setDismissTrackingModeForCameraPositionChange(boolean willAllowToDismiss) {
-    isResetTrackingWithCameraPositionChange = willAllowToDismiss;
+  void resetTrackingModesIfRequired(CameraPosition currentCameraPosition, CameraPosition targetCameraPosition,
+                                    boolean isFromLocation) {
+    resetTrackingModesIfRequired(currentCameraPosition.target != targetCameraPosition.target, false, isFromLocation);
```

Comment:
@tobrun that would mean resetting tracking when zooming, which is an issue ([#9105](https://github.com/mapbox/mapbox-gl-native/issues/9105)).